### PR TITLE
cowsay: 3.03+dfsg2 -> 3.0.4

### PR DESCRIPTION
**Pull request by legendofmiracles**
###### Motivation for this change
This is needed for another PR that I'll soon open:tm: 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by dotlambda — CHANGES_REQUESTED** [review]
You removed cowthink and its man page.

**Comment by dotlambda** [comment]
Why don't we get the source from https://github.com/tnalpgge/rank-amateur-cowsay like [Arch Linux](https://github.com/archlinux/svntogit-packages/blob/packages/cowsay/trunk/PKGBUILD)?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124394` at ed0534b6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>cowsay</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl1` is a deprecated license, please check if project uses `gpl1Plus` or `gpl1Only` and change `meta.license` accordingly.

Near pkgs/tools/misc/cowsay/default.nix:32:5:
```
   |
32 |     license = licenses.gpl1;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/cowsay/default.nix:21:3:
```
   |
21 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by legendofmiracles** [comment]
> Why don't we get the source from https://github.com/tnalpgge/rank-amateur-cowsay like [Arch Linux](https://github.com/archlinux/svntogit-packages/blob/packages/cowsay/trunk/PKGBUILD)?

This one doesn't seem to have `cowthink`, unless i'm blind

**Comment by legendofmiracles** [comment]
Fixed, i'll add another commit for changing the source.

**Comment by dotlambda** [comment]
Arch Linux uses their `install.sh` script: https://github.com/archlinux/svntogit-packages/blob/packages/cowsay/trunk/PKGBUILD
I think that's a good idea.

**Review by jonringer — COMMENTED** [review]
I see this as one big refactor, do you mind squashing?

```
cowsay: use fetchFromGithub, refactor
```

File: pkgs/tools/misc/cowsay/default.nix [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
-    install -m755 cowsay $out/bin/cowsay
-    ln -s cowsay $out/bin/cowthink
-    install -m644 cowsay.1 $out/man/man1/cowsay.1
-    ln -s cowsay.1 $out/man/man1/cowthink.1
-    install -m644 cows/* -t $out/share/cows/
+    runHook preInstall
+    install -Dm755 cowsay $out/bin/cowsay
+    ln -s $out/bin/cowsay $out/bin/cowthink
+    ln -s cowsay.1 cowthink.1
+    installManPage cowsay.1 cowthink.1
```
Comment (dotlambda) [inline pkgs/tools/misc/cowsay/default.nix:31]: This is the the wrong way round. We should create a symlink in `$out` to reduce closure size.

Comment (legendofmiracles) [reply pkgs/tools/misc/cowsay/default.nix:31]: I can't really get that to work, `installManPage` automatically seems to install the man page in the "man" part, which I don't have a reliable way of accessing (i think?)

If I `install` the man page manually, and then make a symlink, it fails with this error:
```
moving /nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/man to /nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/share/man
Moving /nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/share/man to /nix/store/swj9d21ma3a5jp4gcxf3qn0wd6pmrhaf-cowsay-3.04-man/share/man
rmdir: failed to remove '/nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/share': Directory not empty
```

Comment (dotlambda) [reply pkgs/tools/misc/cowsay/default.nix:31]: If you want to keep multiple outputs (I don't see why) you have to use `$man` instead of `$out`.

Comment (dotlambda) [reply pkgs/tools/misc/cowsay/default.nix:31]: `ln -s cowsay.1 $man/share/man/man1/cowthink.1` should work

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please squash the commits into one with message `cowsay: 3.03+dfsg2 -> 3.0.4`.

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 124394` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cowsay</li>
  </ul>
</details>

**Pull request merged**